lib/widgets/expandable_text_widget.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:catchqr/config/app_ui_config.dart';
+import 'package:catchqr/theme/app_theme.dart';
+import 'package:catchqr/utils/context_extensions.dart';
+import 'package:flutter/material.dart';
+
+class ExpandableTextWidget extends StatelessWidget {
+  final int maxLinesToShow;
+  final String content;
+
+  ExpandableTextWidget({
+    super.key,
+    required this.content,
+    required this.maxLinesToShow,
+  });
+
+  final ValueNotifier<bool> _expanded = ValueNotifier(false);
+
+  @override
+  Widget build(BuildContext context) {
+
+    final TextSpan textSpan = TextSpan(
+      text: content,
+      style: context.textTheme.smallText
+    );
+
+    final TextPainter textPainter = TextPainter(
+      text: textSpan,
+      maxLines: _expanded.value ? null : maxLinesToShow,
```

Comment:
does this refresh in some way? cause it's not inside ValueListenableBuilder? - just asking - don't have enough brain right now for proper analysis

Reply:
No, its not refreshing. This part of code is only to calculate lines.